internal/filterapi/filterconfig.go

```diff
@@ -36,7 +39,7 @@ type Config struct {
 	// the "calculated" cost in the filter metadata at the end of the response body processing.
 	LLMRequestCosts []LLMRequestCost `json:"llmRequestCosts,omitempty"`
 	// ModelNameHeaderKey is the header key to be populated with the model name by the filter.
-	ModelNameHeaderKey string `json:"modelNameHeaderKey"`
+	ModelNameHeaderKey internalapi.ModelNameHeaderKey `json:"modelNameHeaderKey"`
```

Comment:
side note: I was looking to drop this ModelNameHeaderKey configurability from here there's no reason to make it so... let's think about it later ...

Reply:
yeah I also would prefer to delete this and just re-use the value constant. would be a lovely post cleanup!